conjure-java-core/src/main/java/com/palantir/conjure/java/services/dialogue/AsyncGenerator.java

```diff
@@ -160,82 +175,37 @@ private MethodSpec asyncClientImpl(ClassName serviceClassName, EndpointDefinitio
         return asyncClient;
     }
 
-    private void addParameter(CodeBlock.Builder request, String endpointName, ArgumentDefinition param) {
-        param.getParamType().accept(new ParameterType.Visitor<CodeBlock.Builder>() {
+    private CodeBlock generateParam(String endpointName, ArgumentDefinition param) {
+        return param.getParamType().accept(new ParameterType.Visitor<CodeBlock>() {
             @Override
-            public CodeBlock.Builder visitBody(BodyParameterType value) {
-                return request.add(
+            public CodeBlock visitBody(BodyParameterType value) {
+                if (param.getType().accept(TypeVisitor.IS_BINARY)) {
```

Comment:
in our glorious new dialogue future, maybe we can even support aliases of binary ?!?!